src/main/java/cs361/battleships/models/Board.java

```diff
@@ -50,42 +49,71 @@ public boolean placeShip(Ship ship, int x, char y, boolean isVertical) {
 		return true;
 	}
 
-	/*
-	DO NOT change the signature of this method. It is used by the grading scripts.
-	 */
 	public Result attack(int x, char y) {
-		Result attackResult = attack(new Square(x, y));
+
+		// Attack surface ships
+		Result attackResult = attack(ships, new Square(x, y) );
 		attacks.add(attackResult);
+
+		// Attack subsurface ships if laser has been earned
+		if ( laserEarned ) {
+
+			Result subAttackResult = attack( submarines, new Square(x, y) );
+			subAttackResult.setSubmerged();
```

Comment:
this should turn into 
subAttackResult.setSubmerged(true);

Reply:
I agree!